laravel/app/Http/Controllers/PostController.php

```diff
@@ -63,27 +64,27 @@ public function store(Request $request)
         return redirect('top');
     }
 
-  /**
-   * Display the specified resource.
-   *
-   * @param  int  $id
-   * @return \Illuminate\Http\Response
-   */
-  public function show()
-  {
-
-  }
+    /**
+    * Display the specified resource.
+    *
+    * @param  int  $id
+    * @return \Illuminate\Http\Response
+    */
+    public function show()
+    {
+    }
 
-  /**
-   * Show the form for editing the specified resource.
-   *
-   * @param  int  $id
-   * @return \Illuminate\Http\Response
-   */
-  public function edit($id)
-  {
-      //
-  }
+    /**
+    * Show the form for editing the specified resource.
+    *
+    * @param  int  $id
+    * @return \Illuminate\Http\Response
+    */
+    public function edit($id)
+    {
+      $posts = User::find(Auth::id())->posts()->where('id',$id)->first();
```

Comment:
idを使って1件だけ取得する場合にはwhere, firstを組み合わせるよりも適したメソッドがあるので修正してみましょう
(updateの方も同様）

Reply:
また1件だけしか入れない場合は、変数名も単数形のほうが良いです
```suggestion
      $post = User::find(Auth::id())->posts()->where('id',$id)->first();
```

Reply:
シンプルに考えて記述すればよかったですね
 $post = Post::find($id);
こうしました